packages/connections/src/utils/validation-errors.spec.ts

```diff
@@ -0,0 +1,284 @@
+      ['ssl=true', 'tls=true'].forEach((option) => {
+        const connectionInfo: ConnectionInfo = {
+          id: 'connection-test',
+          connectionOptions: {
+            connectionString: `mongodb://myserver.com?${option}`,
+            sshTunnel: {
+              host: 'my-hostname',
+              port: 22,
+              username: 'mongouser',
+              password: 'password',
+            },
+          },
+        };
+        const result = validateConnectionInfoErrors(connectionInfo);
+        expect(result).to.deep.equal([
+          {
+            message:
+              'tlsAllowInvalidHostnames is currently required when connecting via SSH tunnel with TLS enabled',
+          },
+        ]);
+      });
+    });
+  });
+
+  describe('X509', function () {
+    it('should return error if tls or ssl is not enabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'TLS must be enabled in order to use x509 authentication.',
+        },
+      ]);
+    });
+    it('should return error if tls is disabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&tls=false&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'TLS must be enabled in order to use x509 authentication.',
+        },
+      ]);
+    });
+
+    it('should return error if ssl is disabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&ssl=false&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'TLS must be enabled in order to use x509 authentication.',
+        },
+      ]);
+    });
+    it('should not return error if tls is enabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&tls=true&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.be.empty;
+    });
+
+    it('should not return error if ssl is enabled', function () {
```

Comment:
there is another condition to check here:
When neither `ssl` or `tls` are set (they are missing or empty): if the schema is `mongodb+srv` it should not error, if is not mongodb+srv it should error